transformer_engine/pytorch/csrc/extensions.h

```diff
@@ -278,13 +278,13 @@ void fp8_block_scaling_partial_cast(const at::Tensor &inp, at::Tensor out, const
  **************************************************************************************************/
 
 at::Tensor fused_rope_forward(const at::Tensor &input, const at::Tensor &freqs,
-                              const NVTE_QKV_Format qkv_format, const bool interleaved,
-                              const std::optional<at::Tensor> cu_seqlens, const int cp_size,
-                              const int cp_rank);
+                              const at::Tensor &start_positions, const NVTE_QKV_Format qkv_format,
+                              const bool interleaved, const c10::optional<at::Tensor> cu_seqlens,
```

Comment:
Please use `std::optional`. PyTorch upstream is deprecating c10::optional. See https://github.com/pytorch/pytorch/pull/150464. And in https://github.com/NVIDIA/TransformerEngine/pull/1683, I replaced all `c10::optional` with `std::optional`.